openupgradelib/openupgrade_merge_records.py

```diff
@@ -51,28 +51,78 @@ def _change_foreign_key_refs(env, model_name, record_ids, target_record_id,
         except (ProgrammingError, IntegrityError) as error:
             if error.pgcode != UNIQUE_VIOLATION:
                 raise
-            # Fallback on setting each row separately
+            # Check if the table contains an id
             env.cr.execute(
-                """ SELECT id FROM %(table)s
-                    WHERE "%(column)s" in %(record_ids)s """, {
-                        'table': AsIs(table),
-                        'column': AsIs(column),
-                        'record_ids': record_ids})
-            for row in env.cr.fetchall():
-                try:
-                    with env.cr.savepoint():
-                        logged_query(
-                            env.cr,
-                            """
-                            UPDATE %(table)s
-                            SET "%(column)s" = %(target_record_id)s
-                            WHERE id = %(id)s """, {
-                                'id': row[0],
-                                'table': AsIs(table), 'column': AsIs(column),
-                                'target_record_id': target_record_id})
-                except (ProgrammingError, IntegrityError) as error:
-                    if error.pgcode != UNIQUE_VIOLATION:
-                        raise
+                """ SELECT column_name
+                    FROM information_schema.columns
+                    WHERE table_name = '%(table)s' and column_name='id'
+                """, {
+                        'table': AsIs(table)
+                })
+            if not env.cr.fetchone():
+                # There's no id. So this is a many2many table. We need
+                # to remove the record.
```

Comment:
`remove` is misleading here you are not doing that in the code, you are actually replacing with the target record, wich I think is good.